app/models/pawn.rb

```diff
@@ -1,11 +1,42 @@
 class Pawn < Piece
 
+  def valid_move?(x, y)
+    unless super(x, y)
+      return false
+    end
+    x_distance = x_position - x
+    y_distance = y_position - y
+    if occupied?(x, y)
+      if color == 'white'
+        x_distance.abs == 1 && y_distance == 1
+      else
+        (-x_distance).abs == 1 && -y_distance == 1
+      end
+    elsif color == 'white'
+      unless has_moved
```

Comment:
super nitpicky but can you switch the unless and the statements? so they follow:
```
if has_moved
  x_distance == 0 && y_distance == 1
else
  (x_distance == 0 && y_distance == 1) || (x_distance == 0 && y_distance == 2)
end
```

just so the next person reading this doesn't have to context switch a bunch